python_modules/dagster/dagster_tests/execution_tests/pipes_tests/test_subprocess.py

```diff
@@ -608,3 +609,45 @@ def raises(context: OpExecutionContext, pipes_client: PipesSubprocessClient):
         assert len(pipes_msgs) == 2
         assert "successfully opened" in pipes_msgs[0]
         assert "external process pipes closed with exception" in pipes_msgs[1]
+
+
+def test_pipes_expected_materialization():
+    def script_fn():
+        ...
+
+    @asset
+    def missing_results(context: OpExecutionContext, pipes_client: PipesSubprocessClient):
+        with temp_script(script_fn) as script_path:
+            cmd = [_PYTHON_EXECUTABLE, script_path]
+            return pipes_client.run(
+                command=cmd,
+                context=context,
+            ).get_reported_results()
+
+    with pytest.raises(
+        DagsterInvariantViolationError,
+        # less then ideal error message
+        match=r"op 'missing_results' did not yield or return expected outputs {'result'}",
+    ):
```

Comment:
may need to tweak the `set_requires_typed_event_stream` exception content to account for this failure condition, though theres already a lot in that message